common/lib/client/auth.js

```diff
@@ -260,7 +259,13 @@ var Auth = (function() {
 						}
 					}
 					cb(null, body);
-				});
+				};
+				Logger.logAction(Logger.LOG_MICRO, 'Auth.requestToken().tokenRequestCallback', 'Sending; ' + authOptions.authUrl + '; Params: ' + JSON.stringify(authParams));
+				if(authOptions.authMethod && authOptions.authMethod.toLowerCase() === 'post') {
+					Http.postUri(rest, authOptions.authUrl, authHeaders || {}, authParams, {}, authUrlRequestCallback);
```

Comment:
> My argument is that the client may require params and/or headers to support its general processing/validation/routing of the URL

Ok, but for params they can specify this as part of the `authUrl`, and for any headers, they can provide this using `authHeaders`

>> if you wanted to add query params to the URL you could simply change the URL
> That's not what's implemented.

Why not?  `new Ably.Realtime({ authUrl: 'x?param1=x&param2=y' })`

> on re-reading it, spec also says it should be "form-encoded

@SimonWoolf this is simple key value string pairs, so you don't need to worry about Javascript objects.  In typed languages this is not possible, and if someone is stupid enough to add objects to  authParams then they can waste their time trying to figure out why it's not encoded as they expected.

We decide on `form-encoded` because it's the most widely supported